src/avatar/src/Avatar.tsx

```diff
@@ -17,65 +17,84 @@ const initialsStyleClass = css({
-     * The name is used as the hashValue by default.
-     * When dealing with anonymous users you should use the id instead.
-     */
-    hashValue: PropTypes.string,
+export interface AvatarProps extends React.ComponentProps<typeof Box> {
+  /**
+   * The color used for the avatar.
+   * When the value is `automatic`, use the hash function to determine the color.
+   */
+  color: 'automatic' | string
+
+  /**
+   * When true, force show the initials.
+   * This is useful in some cases when using Gravatar and transparent pngs.
+   */
+  forceShowInitials?: boolean
+
+  /**
+   * Function to get the initials based on the name.
+   */
+  getInitials: (name?: string | any, fallback?: string) => string
+
+  /**
+   * The value used for the hash function.
+   * The name is used as the hashValue by default.
+   * When dealing with anonymous users you should use the id instead.
+   */
+  hashValue?: string | number
+
+  /**
+   * When true, render a solid avatar.
+   */
+  isSolid?: boolean
+
+  /**
+   * The name used for the initials and title attribute.
+   */
+  name?: string
+
+  /**
+   * The size of the avatar.
+   */
+  size: number
+
+  /**
+   * When the size is smaller than this number, use a single initial for the avatar.
+   */
+  sizeLimitOneCharacter: number
+
+  /**
+   * The src attribute of the image.
+   * When it's not available, render initials instead.
+   */
+  src?: string
+
+  /**
+   * Theme provided by ThemeProvider.
+   */
+  theme: Theme
+}
 
-    /**
-     * When true, render a solid avatar.
-     */
-    isSolid: PropTypes.bool,
+interface AvatarState {
+  imageHasFailedLoading: boolean
+}
 
-    /**
-     * The color used for the avatar.
-     * When the value is `automatic`, use the hash function to determine the color.
-     */
+class Avatar extends PureComponent<AvatarProps, AvatarState> {
+  static propTypes = {
     color: PropTypes.string.isRequired,
-
-    /**
-     * Function to get the initials based on the name.
-     */
-    getInitials: PropTypes.func,
-
-    /**
-     * When true, force show the initials.
-     * This is useful in some cases when using Gravatar and transparent pngs.
-     */
     forceShowInitials: PropTypes.bool,
-
-    /**
-     * When the size is smaller than this number, use a single initial for the avatar.
-     */
-    sizeLimitOneCharacter: PropTypes.number,
-
-    /**
-     * Theme provided by ThemeProvider.
-     */
-    theme: PropTypes.object.isRequired
+    getInitials: PropTypes.func as PropTypes.Validator<
+      (name?: string | any, fallback?: string) => string
+    >,
+    hashValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    isSolid: PropTypes.bool,
+    name: PropTypes.string,
+    size: PropTypes.number.isRequired,
+    sizeLimitOneCharacter: PropTypes.number.isRequired,
+    src: PropTypes.string,
+    theme: PropTypes.object.isRequired as PropTypes.Validator<Theme>
   }
 
-  static defaultProps = {
+  static defaultProps: Partial<AvatarProps> = {
```

Comment:
Isn't the `defaultProps` type automatically inferred from `PureComponent<AvatarProps, AvatarState>`?
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/588c5b81362cac6a460be54b3d890108c9e0c465/types/react/index.d.ts#L526

Reply:
It's supposed to be sometimes complains about the types still. It might be solved with `as const` though.

Reply:
`as const` should fix any errors with string literal types.